src/background/client.ts

```diff
@@ -265,4 +289,14 @@ export class Client {
             this._loadRoom(message.roomId);
         }
     }
+    private async joinAndConfigureRoom(roomId: string, url: string) {
```

Comment:
I'm not sure I like this approach (sorry 😞 ). Barging into a room and trying to set state events just seems... impolite? The use case for explicitly joining a room (rather than being invited, which is a case already handled, I hope) is some sort of public room that stores highlights. But that room would already be configured, then.

I suggest we split the two actions somehow:
1. Joining a room (this) -- you join an existing room and just assume it has a state event already set (or a create event). Then, the existing `Room` handler of the matrix client should take care of emitting the room, and all is well.
2. "Promoting" a room (new) -- take an existing room, that you're in and have permission to modify via state events -- and send the new state event to it, making it a room for your current URL.